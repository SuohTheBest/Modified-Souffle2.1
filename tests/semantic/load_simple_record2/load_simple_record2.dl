// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

// Test reading of multiple records

.type tuple = [f: number, s: number, t: symbol]


.decl A(x: tuple, y: tuple, z: tuple)
.input A()

.decl B(x: tuple, y: tuple, z: tuple)
B(x, y, z) :- A(x, y, z).

.output B(IO=stdout)
