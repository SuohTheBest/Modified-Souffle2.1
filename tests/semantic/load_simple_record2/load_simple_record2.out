---------------
B
===============
nil	nil	nil
nil	[2, 4, apple]	[2, 4, apple]
[1, 2, meow]	[0, 2, meowMeow]	nil
[1, -3, blue]	nil	[1, 2, meow]
===============
